.pre-commit-config.yaml

```diff
@@ -0,0 +1,23 @@
+repos:
```

Comment:
If we are going to add pre-commit, we will need to add a check to the CI to ensure that all of checks pass before running tests.

Also addition of pre-commit as a requirement.

Reply:
Added to requirements in 8f43657. Added to CI in 2097e2d.